dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/EnrollmentTrackerConverterService.java

```diff
@@ -164,28 +164,19 @@ private ProgramInstance from( TrackerPreheat preheat, Enrollment enrollment, Pro
             enrollment.setStatus( EnrollmentStatus.ACTIVE );
         }
 
-        programInstance.setEndDate( DateUtils.fromInstant( enrollment.getCompletedAt() ) );
```

Comment:
Why is getCompletedAt not taken anymore for setEndDate? Should it already have been only set server side?

Reply:
CompletedAt as CompletedBy are read-only fields.
The previous behaviour was incorrect